torch/nn/parallel/distributed.py

```diff
@@ -35,12 +32,54 @@ class DistributedDataParallel(Module):
     See also: :ref:`distributed-basics` and :ref:`cuda-nn-dataparallel-instead`.
     The same constraints on input as in :class:`torch.nn.DataParallel` apply.
 
-    Creation of this class requires the c10d process group to be already
-    initialized. This class will basically operate on the provided c10d
-    process group.
+    Creation of this class requires that ``torch.distributed`` to be already
+    initialized, by calling :func:`torch.distributed.init_process_group`
+
+    ``DistributedDataParallel`` can be used in the following two ways:
+
+    (1) Single-Process Multi-GPU
+
+    In this case, a single process will be
+    spawned on each host/node and each process will operate on all the GPUs
+    of the node where it's running. To use ``DistributedDataParallel`` in
+    this way, you can simply construct the model as the following:
+
+        >>> torch.distributed.init_process_group(backend="nccl")
+        >>> model = DistributedDataParallel(model) # device_ids will include all GPU devices be default
+
+    (2) Multi-Process Single-GPU
+
+    This is the highly recommended way to use ``DistributedDataParallel``, with
+    multiple processes, each of which operates on a single GPU. This is
+    currently the fastest approach to do data parallel training using PyTorch
+    and applies to both single-node(multi-GPU) and multi-node data
+    parallel training. It is proven to be significantly faster than
+    :class:`torch.nn.DataParallel` for single-node multi-GPU data
+    parallel training.
+
+    Here is how to use it: on each host with N GPUs, you should spawn up N
+    processes, while ensuring that each process invidually works on a single GPU
+    from 0 to N-1. Therefore, it is your job to ensure that your training script
+    operates on a single given GPU by calling:
+
+        >>> torch.cuda.set_device(i)
+
+    where i is from 0 to N-1. In each process, you should refer the following
+    to construct this module:
+
+        >>> torch.distributed.init_process_group(backend='nccl', world_size=4, init_method='...')
+        >>> model = DistributedDataParallel(model, devices_ids=[i], output_device=i)
```

Comment:
Just wanted to notice the parameter name is "device_ids" not "devices_ids"

Reply:
Nice catch, thanks!